### ISSUE TITLE
SQL Translation Error

### ISSUE THREAD

**Issue by amir-hz** [jOOQ/jOOQ#7302]
This is SQL Server queries
### SQL translator incorrectly translates this query:

```sql
declare @ID_ITEM as int
declare @DESCRIPTION as varchar(100)

set @ID_ITEM=(select top 1 (ID_ITEM) 1 as ID_ITEM from ITEM where len(ID_ITEM)<>8 order by ID_ITEM desc)
set @DESCRIPTION='Example of description'


--Save to table--
insert into ITEM (ID_ITEM, ID_CLASS, DESCRIPTION, MANUAL, KP_ITEM, ID_KAT)
values (@ID_ITEM,	'7013',	@DESCRIPTION,	NULL, @ID_ITEM,	0)

insert into ITEM_UNIT ( ID_ITEM, ID_UNIT) values (@ID_ITEM,'713')

insert into FOLLOW (ID_FOLLOW, ID_ITEM, ID_UNIT, DATE_FIRST, FOLLOW, DATE_END) values
('TU',	@ID_ITEM,	'713',	'2017-10-01', 	0,	NULL)

insert into FOLLOW_COMPONENT (ID_COMPONENT, ID_FOLLOW, ID_ITEM, ID_UNIT, DATE_FIRST, FOLLOW, FOLLOW_PERCENT, ID_PARENT)
values(10,	'TU',	@ID_ITEM,	'713',	'2017-10-01', 0,	0,	0)


--View Table after save--
select * from ITEM where ID_ITEM =@ID_ITEM
select * from ITEM_UNIT where ID_ITEM=@ID_ITEM
select * from FOLLOW where ID_ITEM=@ID_ITEM
select * from FOLLOW_COMPONENT where ID_ITEM=@ID_ITEM
```

### Parser output:

```sql
Unexpected content after end of field input: [1:9] declare [*]@ID_ITEM as int
declare @DESCRIPTION as varchar(100)

set @ID_ITEM=(select top 1...
```

**Comment by lukaseder** [comment]
Thank you very much for your report.

There are a set of missing features, currently, to correctly parse your statement batch:

- The ability of terminating a statement by newline, rather than by a semi colon `;` or some other delimiter (due in #7171)
- Support for the `DECLARE` statement (as well as the capability of emulating it in other databases): #7305
- The ability to assign values to a variable #7306
- The ability to translate SQL Server statement batches to cursor returning blocks (e.g. in Oracle): #7138

This might all take a while before we can ship the functionality in jOOQ

**Comment by lukaseder** [comment]
I'm closing this as a duplicate of the various other issues I've created
